File: Data Structures & Algorithms/binary-search/submission-0.go
func search(nums []int, target int) int {
    if len(nums) == 0 {
        return -1
    }
    left:=0
    right:=len(nums)-1
    for left <= right {
        mid:=(right+left)/2
        if nums[mid] == target {
            return mid
        } else if nums[mid] > target {
            right=mid-1
        } else {
            left=mid+1
        }
    }
    return -1
}

File: Data Structures & Algorithms/merge-sorted-array/submission-0.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        nums1[m:] = nums2
        nums_copy = nums1[:]

        def merge_sort(nums):
            if len(nums) <= 1:
                return nums
            mid = len(nums) // 2
            left = merge_sort(nums[:mid])
            right = merge_sort(nums[mid:])
            return just_sort(left, right)

        def just_sort(left, right):
            i, j = 0, 0
            res = []
            while i < len(left) and j < len(right):
                if left[i] <= right[j]:
                    res.append(left[i])
                    i += 1
                else:
                    res.append(right[j])
                    j += 1
            res.extend(left[i:])
            res.extend(right[j:])
            return res

        nums1[:] = merge_sort(nums_copy)